app/src/main/java/com/wednesday/bankingdetails/activity/MainActivity.kt

```diff
@@ -30,8 +35,22 @@ class MainActivity : AppCompatActivity() {
             )
         )
 
-        bankListAdapter = BankListAdapter(bankList)
-        recyclerView.adapter = bankListAdapter
+
+        val bankDatabase = FavouritesDatabase.getDatabase(this)
+
+        GlobalScope.launch {
+            val data = bankDatabase?.favouriteBankDao()?.getAll()
+
+            data?.forEach {
+                bankList.find { bank -> bank.ifsCode == it.ifsc }!!.isFavourite = true
+            }
+
+            withContext(Dispatchers.Main) {
+                bankListAdapter = BankListAdapter(bankList)
```

Comment:
Do not do this. If this thread is slow you will see a blank screen and then you will see the list.

- Always set the adapter. Even if you have to show partial information.
- Trigger the thread
- Set the datasource
- Call reset datasource on the adapter and don't create a new object. This will cause a flicker.

Reply:
Done